ENG-105: mds helm chart

jmsuzuki opened the pull request:
**ENG-105: added the mds helm chart**



**ENG-105: added vscode settings for helm templates**

jmsuzuki committed:
ENG-105: added the mds helm chart

jmsuzuki committed:
ENG-105: added vscode settings for helm templates

jmsuzuki commented on mds/templates/deployment.yaml:
```diff
@@ -0,0 +1,233 @@
+      labels:
+        app: mds
+        environment: {{ .Values.deployment.environment }}
+        mds-environment: {{ .Values.deployment.environment }}
```
old. resolved.

jmsuzuki commented on mds/templates/secrets/boreal-webhook-secret.yaml:
```diff
@@ -0,0 +1,11 @@
+    {{- include "mds.labels" . | nindent 4 }}
+type: Opaque
+stringData:
+  secret: {{ required "Boreal webhook secret is required" (default .Values.secrets.borealWebhook.secret (index .Values.secrets.borealWebhook.data "secret")) | quote }}
```
# Fallback logic: Use .Values.secrets.borealWebhook.secret if defined, otherwise use .Values.secrets.borealWebhook.data["secret"].
  
 data access pattern here is wrong.
 `.Values.secrets.borealWebhook.secret (index .Values.secrets.borealWebhook.data "secret"))` is the correct way to grab hyphenated values.

jmsuzuki commented on mds/templates/_helpers.tpl:
```diff
@@ -0,0 +1,62 @@
+Create the name of the service account to use
+*/}}
+{{- define "mds.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
```
old. resolved

callicles commented on mds/templates/secrets/clickhouse-secret.yaml:
Is this clickhouse cloud connection information?

jmsuzuki replied:
yup!

callicles reviewed (approved):
How do you plan to deploy Kafka / Clickhouse / redis / temporal ?

I am not sure if those should get deployed, maybe outside or inside helm?

jmsuzuki commented:
> How do you plan to deploy Kafka / Clickhouse / redis / temporal ?
> 
> I am not sure if those should get deployed, maybe outside or inside helm?

I have a repo for the boreal byoc deployment examples AND another repo for the terraform boreal byoc module.
https://github.com/514-labs/boreal-byoc-examples
https://github.com/514-labs/terraform-boreal-byoc

That repo will eventually contain terraform and pulumi directories with various deployment examples given.
To begin, I'll be doing the example in pulumi first, but from a terraform converted module for the mds deployment.

The terraform module will pull in the helm chart and deploy just that.

Kafka, Clickhouse, redis, and temporal will be defined and deployed by the examples repo.
As we start making moose more configurable in terms of the provided services, those examples should be relatively easy to add.

jmsuzuki commented on mds/values.yaml:
```diff
@@ -0,0 +1,161 @@
+
+# Staging-specific environment variables
+staging:
+  github514CommercialRepoId: ""
```
remove this...

jmsuzuki commented on mds/values.yaml:
```diff
@@ -0,0 +1,161 @@
+  borealWebhook:
+    name: "boreal-webhook-secret"  # Updated to match cluster
+    data: {}
+    secret: ""
```
each byoc customer should have their own webhook secret.

callicles commented:
<!-- linear:replyTo:3053610882 -->

Why not include the dependencies in the terraform or helm with some configuration?<br><br>In a BYOC env we are supposed to manage those for the customer. Not have the customer manage that themselves

jmsuzuki committed:
feat: convert to helm chart repository with mono-repo structure

- Restructured repository to support multiple charts
- Moved mds chart to charts/ directory
- Added helper scripts for packaging and publishing
- Set up GitHub Actions for CI/CD
- Added comprehensive documentation
- Fixed template issues in mds chart
- Configured .gitignore for helm repository

jmsuzuki committed:
fix: improve publish script to handle branch switching correctly

- Copy files to temp directory before switching branches
- Add error handling for stash operations
- Clean up temp directory after use

jmsuzuki committed:
fix: package script now fetches existing index.yaml from gh-pages

- Prevents losing references to older chart versions
- Fetches existing index.yaml from gh-pages before merging
- Ensures all packaged versions remain accessible

jmsuzuki commented:
> Why not include the dependencies in the terraform or helm with some configuration?
> 
> In a BYOC env we are supposed to manage those for the customer. Not have the customer manage that themselves

I guess I had a slightly different expectation. I expected the customer to bring their own everything to bring your own cloud.
We provide MDS to deploy in their environment and then they provide the various credentials and connections strings to connect to and manage their services (clickhouse, temporal, kafka/redpanda).

The examples repo is intended to provide customers examples on how to deploy various byoc types.
As the number of deployment options grows, the number of examples in that repo will grow and this repo / chart can stay clean / focused on the services in deploys.

This is an interesting question though and I definitely considered it, especially as I deployed temporal.
When deploying temporal's helm chart, it'll even deploy an observability stack if you like.

```
byoc-temporal            elasticsearch-master-0                          1/1     Running     0             14h
byoc-temporal            temporal-c10b0e49-admintools-85ddb5cc56-b4kwv   1/1     Running     0             14h
byoc-temporal            temporal-c10b0e49-cassandra-0                   1/1     Running     0             14h
byoc-temporal            temporal-c10b0e49-frontend-cc76c7c8d-v8v9n      1/1     Running     0             12h
byoc-temporal            temporal-c10b0e49-history-78595df7fd-dvv59      1/1     Running     2 (14h ago)   14h
byoc-temporal            temporal-c10b0e49-matching-64d9c6658c-rhqdr     1/1     Running     2 (14h ago)   14h
byoc-temporal            temporal-c10b0e49-schema-2-blx4f                0/1     Completed   0             14h
byoc-temporal            temporal-c10b0e49-web-c97567875-9zbxr           1/1     Running     0             14h
byoc-temporal            temporal-c10b0e49-worker-668bddd684-59mm9       1/1     Running     0             12h
```

EDIT: I also dislike managing too many things in a single stack / deployment.
Deployments and Upgrades are faster and less risky when moving small.

jmsuzuki commented:
merging this as is for now... Nico and I chatted about next steps for BYOC.
we will be driving down HD's path for BYOC, using our redis, no kafka, and ch and temporal deployed in k8s.
